src/formats/farbfeld.zig

```diff
@@ -0,0 +1,119 @@
+const std = @import("std");
+const builtin = @import("builtin");
+const Allocator = std.mem.Allocator;
+const assert = std.debug.assert;
+const ImageUnmanaged = @import("../ImageUnmanaged.zig");
+const FormatInterface = @import("../FormatInterface.zig");
+const color = @import("../color.zig");
+const utils = @import("../utils.zig");
+const buffered_stream_source = @import("../buffered_stream_source.zig");
+
+pub const ReadError = ImageUnmanaged.ReadError;
+pub const WriteError = ImageUnmanaged.WriteError || error{BigDimensions};
+
+pub const Header = extern struct {
+    width: u32 align(1),
+    height: u32 align(1),
+
+    pub const size = 16;
+    const width_size = 4;
+    const height_size = 4;
+    const magic_value: [8]u8 = "farbfeld".*;
+
+    fn encode(header: Header) [size]u8 {
+        var result: [size]u8 = undefined;
+        @memcpy(result[0..8], &magic_value);
+        std.mem.writeInt(u32, result[8..12], header.width, .big);
+        std.mem.writeInt(u32, result[12..16], header.height, .big);
+        return result;
+    }
+    comptime {
+        assert(@sizeOf(Header) + @sizeOf(@TypeOf(magic_value)) == size);
+    }
+};
+
+pub const Farbfeld = struct {
+    header: Header,
+    pixels: []color.Rgba64,
+
+    pub fn formatInterface() FormatInterface {
+        return FormatInterface{
+            .format = format,
+            .formatDetect = formatDetect,
+            .readImage = readImage,
+            .writeImage = writeImage,
+        };
+    }
+
+    pub fn format() ImageUnmanaged.Format {
+        return ImageUnmanaged.format.farbfeld;
+    }
+    /// Taken a stream, Returns true if and only if the stream contains the magic value "fabfeld"
+    pub fn formatDetect(stream: *ImageUnmanaged.Stream) ReadError!bool {
+        var magic_buffer: [Header.magic_value.len]u8 = undefined;
+        const bytes_read = try stream.read(magic_buffer[0..]);
+        return bytes_read == Header.magic_value.len and
+            std.mem.eql(u8, magic_buffer[0..], Header.magic_value[0..]);
+    }
+    pub fn read(allocator: std.mem.Allocator, stream: *ImageUnmanaged.Stream) ReadError!Farbfeld {
```

Comment:
This should be:
`pub fn read(self: *Farbfeld, allocator: std.mem.Allocator, stream: *ImageUnmanaged.Stream) ImageUnmanaged.ReadError!color.PixelStorage`